src/server.rs

```diff
@@ -24,16 +25,35 @@ pub async fn healthz_handler(
 pub enum MetadataMessage {
     TrackTable(QualifiedTable),
     UntrackTable(QualifiedTable),
-    // NOTE: args will only be `null` in this case
+    // NOTE: args will be `null` for `export_metadata`
     ExportMetadata,
+    ImportMetadata(Metadata),
 }
 
-#[derive(Serialize)]
+#[derive(Serialize, Debug, Clone)]
 pub struct MetadataSuccess {
     success: bool,
     message: String,
 }
 
+fn make_metadata_success_response(message: &str) -> actix_web::HttpResponse {
+    actix_web::web::HttpResponse::Ok().json(MetadataSuccess {
+        success: true,
+        message: String::from(message),
+    })
+}
+
+fn make_metadata_error_response(status: StatusCode, err_message: &str) -> actix_web::HttpResponse {
+    actix_web::web::HttpResponse::build(status).json(ErrorResponse {
+        error: String::from(err_message),
+    })
+}
+
+fn bad_request_response(err_message: &str) -> actix_web::HttpResponse {
+    // NOTE: probably should be making use of currying here
+    make_metadata_error_response(StatusCode::BAD_REQUEST, err_message)
+}
+
```

Comment:
We can follow in the footsteps of `MetadataMessage`, use an enum to represent the Metadata response and impl Responder for it. So we can return it in any function that returns impl Responder like. So we will replace this
```suggestion
#[derive(Serialize, Debug, Clone)]
pub enum MetadataResponse {
    Success(String),
    Data(Metadata),
    Error(String),
}

impl Responder for MetadataResponse {
    type Error = actix_web::Error;
    type Future = actix_web::HttpResponse;

    fn respond_to(self, _: &actix_web::HttpRequest) -> Self::Future {
        match self {
            MetadataResponse::Error(err) => actix_web::HttpResponse::build(StatusCode::BAD_REQUEST)
                .json(json!({ "error": err })),
            MetadataResponse::Success(msg) => {
                actix_web::HttpResponse::Ok().json(json!({"success": true, "message": msg}))
            }
            MetadataResponse::Data(metadata) => actix_web::web::HttpResponse::Ok().json(metadata),
        }
    }
}
```